src/v/archival/ntp_archiver_service.cc

```diff
@@ -502,6 +502,7 @@ void ntp_archiver::update_probe() {
     const auto& man = manifest();
 
     _probe->segments_in_manifest(man.size());
+    _probe->segments_metadata_bytes(man.segments_metadata_bytes());
```

Comment:
Looks like update_probe() is only called from the segment upload loop: this will give us zero stats on followers, and stale stats on nodes that used to be leaders, right?

I believe metrics can be constructed with a lambda for fetching gauges on demand, which might be more robust here than trying to remember to call update_probe in all the right places.

(Clearly this issue already exists for segments_in_manifest too)

Reply:
Nice catch. I moved the metric to accommodate. Will put up a separate PR to address segments_in_manifest and co.